FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Mutual_Auth/DemoTasks/MutualAuthMQTTExample.c

```diff
@@ -546,6 +544,9 @@ static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( NetworkCredent
             pxNetworkCredentials->pAlpnProtos = AWS_IOT_MQTT_ALPN;
             pxNetworkCredentials->alpnProtosLen = AWS_IOT_MQTT_ALPN_LENGTH;
         #endif
+    #else
+        /* SNI needs to be disabled for an MQTT broker that has no hostname. */
```

Comment:
Should the SNI be _always_ disabled for non-AWS IoT brokers?

Reply:
As it says, when there is no hostname. You cannot use SNI when the broker only has an IP address and no DNS.

Reply:
We shouldn't make an assumption that a non-AWS IoT broker being used by the user has no hostname

Reply:
If the customer follows instructions for setting up a local Mosquitto server then when they plug in their local server in the config file, the demo will fail. Disabling it by default will make sure those instructions work. They will know how to enable it for their own application if need be. That is why the comment is there.

Reply:
I will just wrap that assignment in the comment then